doc/en/usage.rst

```diff
@@ -261,8 +261,12 @@ Here is the full list of available characters that can be used:
  - ``X`` - xpassed
  - ``p`` - passed
  - ``P`` - passed with output
+
+Special characters for (de)selection of groups:
+
  - ``a`` - all except ``pP``
  - ``A`` - all
+ - ``N`` - none, this can be used to display nothing (since ``fE`` is the default)
 
 More than one character can be used, so for example to only see failed and skipped tests, you can execute:
 
```

Comment:
This could also somehow mention that the effective default is `wfE` without `--disable-warnings` actually?  (https://github.com/pytest-dev/pytest/issues/5066)

Reply:
Good point, that got me thinking and I think `-rw` should be removed from the docs.

Since we have merged `pytest-warnings` (which seems like decades ago), `-rw` is not really a thing, because independently of what you put in `-r` (except `N` now in this PR) warnings are always displayed:

No `-r` option:

```
λ pytest .tmp\test-reporting.py
...
========================= warnings summary ==========================
test-reporting.py::test_warn
  D:\projects\pytest\.tmp\test-reporting.py:28: UserWarning: some warning
    warnings.warn(UserWarning('some warning'))

-- Docs: https://docs.pytest.org/en/latest/warnings.html
====================== short test summary info ======================
FAILED .tmp\test-reporting.py::test_fail[0] - assert 0
FAILED .tmp\test-reporting.py::test_fail[1] - assert 0
FAILED .tmp\test-reporting.py::test_fail[2] - assert 0
=== 3 failed, 4 passed, 3 skipped, 3 xfailed, 1 warning in 0.11s ====
```

Some `-r` combination (note the missing `w`):

```
λ pytest .tmp\test-reporting.py -rXfs
...
========================= warnings summary ==========================
test-reporting.py::test_warn
  D:\projects\pytest\.tmp\test-reporting.py:28: UserWarning: some warning
    warnings.warn(UserWarning('some warning'))

-- Docs: https://docs.pytest.org/en/latest/warnings.html
====================== short test summary info ======================
FAILED .tmp\test-reporting.py::test_fail[0] - assert 0
FAILED .tmp\test-reporting.py::test_fail[1] - assert 0
FAILED .tmp\test-reporting.py::test_fail[2] - assert 0
SKIPPED [3] test-reporting.py:15: nas
=== 3 failed, 4 passed, 3 skipped, 3 xfailed, 1 warning in 0.12s ====
```

So the only way to skip the *warnings summary* section is by using either `-rN` or `--disable-warnings`, as of now.

Warnings being controlled in `-r` is historical left-over, so much that *warnings summary* and *short test summary* are separate today.

I propose then:

* Remove `-rw` from the docs, given it doesn't do anything (or mention that it is there for historical reasons but has no effect).
* Make `-rN` not affect warnings at all, warnings can already be disabled with `--disable-warnings`. No backward compatibility here to consider because `N` is new, and the only way to hide warnings was with `--disable-warnings`. This also makes `-rN` the exact behavior as before (hide *short test summary*, but *warnings summary* is still shown).

Reply:
@nicoddemus have you looked at https://github.com/pytest-dev/pytest/issues/5066 again?
There we wanted something like `-r-w` and remove/deprecate the option completely even.

Reply:
Ahh I didn't, even thought you mentioned it, my bad.

I've re-read that thread, and I still think what I wrote above holds better, specially because warnings and short summaries are reported and dealt separately. It also means we wouldn't need to deprecate anything really.

Reply:
Ahh I didn't, even thought you mentioned it, my bad.

I've re-read that thread, and I still think what I wrote above holds better, specially because warnings and short summaries are reported and dealt separately. It also means we wouldn't need to deprecate anything really.